sql/connect/server/src/main/scala/org/apache/spark/sql/connect/pipelines/PipelinesHandler.scala

```diff
@@ -0,0 +1,288 @@
+   * @param responseObserver
+   *   The response observer where the response will be sent
+   * @param sparkSession
+   *   The spark session
+   * @param transformRelationFunc
+   *   Function used to convert a relation to a LogicalPlan. This is used when determining the
+   *   LogicalPlan that a flow returns.
+   * @return
+   *   The response after handling the command
+   */
+  def handlePipelinesCommand(
+      sessionHolder: SessionHolder,
+      cmd: proto.PipelineCommand,
+      responseObserver: StreamObserver[ExecutePlanResponse],
+      sparkSession: SparkSession,
+      transformRelationFunc: Relation => LogicalPlan): PipelineCommandResult = {
+    // Currently most commands do not include any information in the response. We just send back
+    // an empty response to the client to indicate that the command was handled successfully
+    val defaultResponse = PipelineCommandResult.getDefaultInstance
+    cmd.getCommandTypeCase match {
+      case proto.PipelineCommand.CommandTypeCase.CREATE_DATAFLOW_GRAPH =>
+        val createdGraphId = createDataflowGraph(cmd.getCreateDataflowGraph)
+        PipelineCommandResult
+          .newBuilder()
+          .setCreateDataflowGraphResult(
+            PipelineCommandResult.CreateDataflowGraphResult.newBuilder
+              .setDataflowGraphId(createdGraphId)
+              .build())
+          .build()
+      case proto.PipelineCommand.CommandTypeCase.DROP_DATAFLOW_GRAPH =>
+        logInfo(s"Drop pipeline cmd received: $cmd")
+        DataflowGraphRegistry.dropDataflowGraph(cmd.getDropDataflowGraph.getDataflowGraphId)
+        defaultResponse
+      case proto.PipelineCommand.CommandTypeCase.DEFINE_DATASET =>
+        logInfo(s"Define pipelines dataset cmd received: $cmd")
+        defineDataset(cmd.getDefineDataset, sparkSession)
+        defaultResponse
+      case proto.PipelineCommand.CommandTypeCase.DEFINE_FLOW =>
+        logInfo(s"Define pipelines flow cmd received: $cmd")
+        defineFlow(cmd.getDefineFlow, transformRelationFunc, sparkSession)
+        defaultResponse
+      case proto.PipelineCommand.CommandTypeCase.START_RUN =>
+        logInfo(s"Start pipeline cmd received: $cmd")
+        startRun(cmd.getStartRun, responseObserver, sessionHolder)
+        defaultResponse
+      case proto.PipelineCommand.CommandTypeCase.DEFINE_SQL_GRAPH_ELEMENTS =>
+        logInfo(s"Register sql datasets cmd received: $cmd")
+        defineSqlGraphElements(cmd.getDefineSqlGraphElements, sparkSession)
+        defaultResponse
+      case other => throw new UnsupportedOperationException(s"$other not supported")
+    }
+  }
+
+  private def createDataflowGraph(cmd: proto.PipelineCommand.CreateDataflowGraph): String = {
+    val defaultCatalog = Option
+      .when(cmd.hasDefaultCatalog)(cmd.getDefaultCatalog)
+      .getOrElse {
+        logInfo(
+          s"No default catalog was supplied. Falling back to the session catalog `spark_catalog`).")
+        "spark_catalog"
```

Comment:
Do we actually want to fallback to this catalog? Or to the one that is currently the default? The same question applies to the `defaultDatabase`...

Reply:
Updated to fallback to the current one